### PR TITLE
[Fixes #2545] RedirectToRouteResult doesn't use RouteName property wh…

### LINKED ISSUE #2545: RedirectToRouteResult doesn't use RouteName property when calculating destination URL
The `Controller.RedirectToRoute()` method creates a `RedirectToRouteResult` instance and passes in the `routeName` and `routeValues` parameters to it, which are assigned to the `RedirectToRouteResult.RouteName` and `RedirectToRouteResult.RouteValues` properties, respectively.

However, on [line 48](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs#L48) in the `RedirectToRouteResult.ExecuteResult()` method, the `RouteName` property is never used to calculate the destination URL; it is essentially ignored.

I would have expected that `Controller.RedirectToRoute()` method to consider my provided `routeName` when calculating the destination URL.

The workaround for now, is pretty simple: just calculate my own destination URL using the controller's UrlHelper, but since `Controller.RedirectToRoute()` didn't behave as I had expected it to, I'm filing an issue for it.

### PR DESCRIPTION
**Pull request by kichalla**
…en calculating destination URL

@rynowak @Eilon

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public override void ExecuteResult([NotNull] ActionContext context)
         {
             var urlHelper = GetUrlHelper(context);
 
-            var destinationUrl = urlHelper.RouteUrl(RouteValues);
+            var destinationUrl = urlHelper.RouteUrl(RouteName, RouteValues);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs]: Are there unit tests for this class? should they be updated/added?

Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs]: Thought the functional tests should be sufficient but yeah there are unit tests for this class...added a new test now...

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/RedirectToRouteResult.cs]: Functional tests are harder to debug and take more time to write. They are great when we want to capture a user-focused scenario or complex integration point.

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
Commit: 4bed2e6f2b3e775ee58deb82ac584210c3e41ed6

**Pull request closed without merging**